draft-ietf-httpbis-digest-headers.md

```diff
@@ -335,49 +335,49 @@ some algorithms, one or more parameters can be supplied.
 ~~~
 
 The BNF for "parameter" is defined in Section 5.4.1.4 of
-{{SEMANTICS}}. All digest-algorithm values are case-insensitive.
+{{SEMANTICS}}. All digest-algorithm values are case-insensitive but SHOULD be used in lowercase.
 
 The Internet Assigned Numbers Authority (IANA) acts as a registry for
 digest-algorithm values.
 The registry contains the tokens listed below.
 
 Some algorithms, although registered, have since been found vulnerable:
-the MD5 algorithm MUST NOT be used due to collision attacks [CMU-836068]
-and the SHA algorithm MUST NOT be used due
+the "md5" algorithm MUST NOT be used due to collision attacks [CMU-836068]
+and the "sha" algorithm MUST NOT be used due
```

Comment:
Unfortunately due to the overloading of algorithm and the way this section is structured, the lowercasing of these values seems inconsistent with your stated aims. A few ways to fix this come to mind:

1) Unwind the change and leave them as uppercase
2) 

```
the "md5" token MUST NOT be used due to collision attacks in the MD5 algorithm [link], the "sha" token MUST NOT be used due to collision attacks [link]
```

3) 
```
the MD5 algorithm (token "md5") MUST NOT be used ...
```

I have a slight preference for option 1.

Reply:
I interpreted the following as digest-algorithms because the sentence starts with

> Some algorithms, although registered

Rewording like below should fix imho

> Some digest-algorithms, although registered

OTOH to reference algorithm names instead of tokens, we should use SHA-1

```suggestion
the MD5 algorithm MUST NOT be used due to collision attacks [CMU-836068]
and the SHA-1 algorithm MUST NOT be used due
```